Zilon.Core/Zilon.Core/Tactics/GameLoop.cs

```diff
@@ -1,87 +1,137 @@
+            var actorsWithoutTasks = GetActorsWithoutTasks(_sectorManager.CurrentSector.ActorManager, _taskDict);
+            await GenerateActorTasksAndPutInDictAsync(actorsWithoutTasks, _taskDict).ConfigureAwait(false);
 
-            var firstIsHumanPlayer = actorsQueue.FirstOrDefault()?.Owner is HumanPlayer;
-            if (!firstIsHumanPlayer && actorsQueue.Any(x => x.Owner is HumanPlayer))
+            ProcessTasks(_taskDict);
+            _turnCounter++;
+            if (_turnCounter >= 1000)
             {
-                throw new InvalidOperationException("Первым должен быть персонаж, которым управляет актёр");
+                _turnCounter = 1000 - _turnCounter;
+
+                _sectorManager.CurrentSector.Update();
+
+                Updated?.Invoke(this, new EventArgs());
             }
+        }
 
-            foreach (var actor in actorsQueue)
+        private void ProcessTasks(IDictionary<IActor, TaskState> taskDict)
+        {
+            var states = taskDict.Values;
+            foreach (var state in states)
             {
-                if (actor.Person.CheckIsDead())
+                if (state.Actor.Person.GetModuleSafe<ISurvivalModule>()?.IsDead == true)
                 {
+                    // Персонажи, у которых есть модуль выживания, могут умереть.
+                    // Мертвые персонажы не выполняют задач.
+                    // Их задачи игнорируем, т.к. задачи могут выполниться после смерти персонажа,
+                    // что противоречит логике задач.
                     continue;
                 }
 
-                ProcessActor(actor);
+                state.UpdateCounter();
+
+                if (state.TaskIsExecuting)
+                {
+                    state.Task.Execute();
+                    state.TaskSource.ProcessTaskExecuted(state.Task);
+                }
             }
 
-            _sectorManager.CurrentSector.Update();
+            // удаляем выполненные задачи.
+            foreach (var taskStatePair in taskDict.ToArray())
+            {
+                var state = taskStatePair.Value;
+                if (state.TaskComplete)
+                {
+                    taskDict.Remove(taskStatePair.Key);
+                    state.TaskSource.ProcessTaskComplete(state.Task);
+                    continue;
+                }
 
-            Updated?.Invoke(this, new EventArgs());
+                if (taskStatePair.Key.Person.GetModuleSafe<ISurvivalModule>()?.IsDead == true)
+                {
+                    // Персонажи, у которых есть модуль выживания, могут умереть.
+                    // Мертвые персонажы не выполняют задач.
+                    // Их задачи можно прервать, потому что:
+                    // 1. Возможна ситуация, когда мертвый персонаж всё еще выполнить действие.
+                    // 2. Экономит ресурсы.
+                    taskDict.Remove(taskStatePair.Key);
+                    continue;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant jump.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5544638)